pkg/controller/sdk/finalizers_accessor.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sdk
+
+import (
+	"reflect"
+)
+
+// FinalizersAccessor is the interface for a Resource that implements the getter and setting for
+// accessing its Finalizer set.
+// +k8s:deepcopy-gen=true
+type FinalizersAccessor interface {
+	GetFinalizers() interface{}
+	SetFinalizers(finalizers interface{})
+}
+
+// NewReflectedFinalizersAccessor uses reflection to return a FinalizersAccessor to access the field
+// called "Finalizers".
```

Comment:
Mention this might return `nil`.

Reply:
Changed it to return (FinalizersAccessor, error) instead. If error == nil, then FinalizersAccessor != nil. I think this is standard Go style, do you want me to document it?

Reply:
No, that seems fine. I think returning a `nil` interface is fine, too (and reasonable Go style), but documentation or explicit error helps to make it obvious that `nil` is a possible return value.